console/console-init/ui/src/modules/connection-detail/containers/ConnectionLinksToolbarContainer/ConnectionLinksToolbarContainer.tsx

```diff
@@ -0,0 +1,380 @@
+      setNameOptions([
+        <SelectOption
+          value={TypeAheadMessage.MORE_CHAR_REQUIRED}
+          isDisabled={true}
+        />
+      ]);
+    } else {
+      onChangeNameInput(input).then(data => {
+        const list = data;
+        const options = list
+          ? list.map((object, index) => (
+              <SelectOption
+                disabled={object.isDisabled}
+                key={index}
+                value={object.value}
+              />
+            ))
+          : [];
+        if (options && options.length > 0) {
+          setNameOptions(options);
+        } else {
+          setNameOptions([
+            <SelectOption
+              value={TypeAheadMessage.NO_RESULT_FOUND}
+              isDisabled={true}
+            />
+          ]);
+        }
+      });
+    }
+    const options = [
+      <SelectOption
+        value={TypeAheadMessage.MORE_CHAR_REQUIRED}
+        key="1"
+        isDisabled={true}
+      />
+    ];
+    return options;
+  };
+  const onAddressSelect = (e: any, selection: SelectOptionObject) => {
+    setAddressSelected(selection.toString());
+  };
+  const onAddressClear = () => {
+    setAddressSelected(undefined);
+    setAddressInput(undefined);
+  };
+
+  const onChangeAddressInput = async (value: string) => {
+    const response = await client.query<IConnectionLinksAddressSearchResponse>({
+      query: RETURN_ALL_CONNECTION_LINKS_FOR_ADDRESS_SEARCH(
+        connectionName,
+        namespace,
+        value.trim()
+      ),
+      fetchPolicy: FetchPolicy.NETWORK_ONLY
+    });
+    if (
+      response &&
+      response.data &&
+      response.data.connections &&
+      response.data.connections.connections &&
+      response.data.connections.connections.length > 0 &&
+      response.data.connections.connections[0].links &&
+      response.data.connections.connections[0].links.links &&
+      response.data.connections.connections[0].links.links.length > 0
+    ) {
+      const obtainedList = response.data.connections.connections[0].links.links.map(
+        (link: any) => {
+          return link.spec.address;
+        }
+      );
+      //get list of unique records to display in the select dropdown based on total records and 100 fetched objects
+      const filteredAddressOptions = getSelectOptionList(
+        obtainedList,
+        response.data.connections.connections[0].links.total
+      );
+      if (filteredAddressOptions.length > 0) return filteredAddressOptions;
+    }
+  };
+
+  const onAddressFilter = (e: any) => {
+    const input = e.target.value && e.target.value.trim();
+    setAddressInput(input);
+    setAddressOptions(undefined);
+    if (input.trim().length < TYPEAHEAD_REQUIRED_LENGTH) {
+      setAddressOptions([
+        <SelectOption
+          value={TypeAheadMessage.MORE_CHAR_REQUIRED}
+          isDisabled={true}
+        />
+      ]);
+    } else {
+      onChangeAddressInput(input).then(data => {
+        const list = data;
```

Comment:
same kind of code/logic used multiple files. is it possible to create generic function for all so that we can reuse the logic.